File: content/pymongo-driver/v4.7/source/includes/write/delete.py
# start-delete-one
query_filter = { "name": "Ready Penny Inn" }

result = restaurants.delete_one(query_filter)
# end-delete-one

# start-delete-many
query_filter = { "borough": "Brooklyn" }

result = restaurants.delete_many(query_filter)
# end-delete-many

# start-delete-options
query_filter = { 'name': {'$regex': 'Mongo' }}

result = restaurants.delete_many(query_filter, comment="Deleting Mongo restaurants")
# end-delete-options

File: content/rust/v3.3/source/includes/fundamentals/code-snippets/crud/delete.rs
use bson::{ Document, bson };
use mongodb::{ bson::doc, options::{ DeleteOptions, Hint }, Client, Collection };
use std::{ env };

#[tokio::main]
async fn main() -> mongodb::error::Result<()> {
    let uri = "<connection string>";
    let client = Client::with_uri_str(uri).await?;

    let my_coll: Collection<Document> = client.database("db").collection("inventory");

    // begin-delete-one
    let filter = doc! { "item": "placemat" };

    let res = my_coll.delete_one(filter).await?;
    println!("Deleted documents: {}", res.deleted_count);
    // end-delete-one

    // begin-delete-many
    let filter = doc! { "category": "garden" };
    let hint = Hint::Name("_id_".to_string());

    let res = my_coll
        .delete_many(filter)
        .hint(hint)
        .await?;
    println!("Deleted documents: {}", res.deleted_count);
    // end-delete-many

    // begin-options
    let res = my_coll
        .delete_one(filter)
        .comment(bson!("hello!"))
        .await?;
    // end-options

    Ok(())
}
